### PR TITLE
Add scale down test and loop count for autoscale

### PR DESCRIPTION
**Pull request by alyssa1303**
Add scale down and loop count for autoscale test to perform the test multiple times with the same set of resources
Current configs:
- scale up timeout: 15m
- scale down timeout: 15m
- loop count: 5
- cool down time between execution: 30s

### CODE REVIEW

File: scenarios/perf-eval/nap-c4n10p100/kubernetes/karpenter_nodepool.aws.yml [hunk 1]
```diff
@@ -3,6 +3,13 @@ kind: NodePool
 metadata:
   name: default
 spec:
+  disruption:
+    consolidationPolicy: WhenEmptyOrUnderutilized
+    consolidateAfter: 1m
```
Comment (anson627) [inline scenarios/perf-eval/nap-c4n10p100/kubernetes/karpenter_nodepool.aws.yml:8]: does it mean karpenter wait for 1m before deleting empty nodes?

File: modules/python/clusterloader2/autoscale/config/config.yaml [hunk 2]
```diff
@@ -37,22 +39,69 @@ steps:
...
+  - Identifier: WaitForNodesDown {{$i}}
+    Method: WaitForNodes
+    Params:
+      action: start
+      minDesiredNodeCount: 0
+      maxDesiredNodeCount: 0
+      labelSelector: {{$nodeLabelSelector}}
+      timeout: {{$scaleDownTimeout}}
+      refreshInterval: {{$refreshInterval}}
+  - Identifier: WaitForRunningPodsDown {{$i}}
```
Comment (anson627) [inline modules/python/clusterloader2/autoscale/config/config.yaml:92]: maybe WaitForRunningPodsDown before WaitForNodesDown?

File: modules/python/clusterloader2/autoscale/config/config.yaml [hunk 3]
```diff
@@ -4,11 +4,12 @@
...
 {{$minNodeCount := DefaultParam .CL2_MIN_NODE_COUNT 10}}
 {{$maxNodeCount := DefaultParam .CL2_MAX_NODE_COUNT 20}}
-{{$nodeMeasurementTimeout := DefaultParam .CL2_NODE_MEASUREMENT_TIMEOUT "30m"}}
-{{$nodeRefreshInterval := DefaultParam .CL2_NODE_REFRESH_INTERVAL "5s"}}
 {{$podLabelSelector := DefaultParam .CL2_POD_LABEL_SELECTOR "app = inflate"}}
-{{$podMeasurementTimeout := DefaultParam .CL2_POD_MEASUREMENT_TIMEOUT "30m"}}
-{{$podRefreshInterval := DefaultParam .CL2_POD_REFRESH_INTERVAL "5s"}}
+{{$scaleUpTimeout := DefaultParam .CL2_SCALE_UP_TIMEOUT "30m"}}
+{{$scaleDownTimeout := DefaultParam .CL2_SCALE_DOWN_TIMEOUT "10m"}}
+{{$refreshInterval := DefaultParam .CL2_REFRESH_INTERVAL "5s"}}
```
Comment (anson627) [inline modules/python/clusterloader2/autoscale/config/config.yaml:10]: just a question, does it mean we check the pods/nodes cache every 5 seconds?

File: modules/python/clusterloader2/autoscale/config/config.yaml (hunk 2 above)
Comment (alyssa1303) [reply modules/python/clusterloader2/autoscale/config/config.yaml:92]: Sure updated

File: modules/python/clusterloader2/autoscale/config/config.yaml (hunk 3 above)
Comment (alyssa1303) [reply modules/python/clusterloader2/autoscale/config/config.yaml:10]: That's right

**Review by anson627 — APPROVED** [review]

File: scenarios/perf-eval/nap-c4n10p100/kubernetes/karpenter_nodepool.azure.yml [hunk 4]
```diff
@@ -3,28 +3,33 @@ kind: NodePool
...
-        - on-demand
-      - key: karpenter.azure.com/sku-family
-        operator: In
-        values:
-        - D
-      - key: karpenter.azure.com/sku-cpu
-        operator: In
-        values:
-        - "4"
+        - key: kubernetes.io/arch
```
Comment (anson627) [inline scenarios/perf-eval/nap-c4n10p100/kubernetes/karpenter_nodepool.azure.yml:16]: tab?

**Pull request merged**